codegen-server/python/src/main/kotlin/software/amazon/smithy/rust/codegen/server/python/smithy/generators/PythonServerStructureGenerator.kt

```diff
@@ -0,0 +1,111 @@
+import software.amazon.smithy.rust.codegen.rustlang.documentShape
+import software.amazon.smithy.rust.codegen.rustlang.render
+import software.amazon.smithy.rust.codegen.rustlang.rust
+import software.amazon.smithy.rust.codegen.rustlang.rustBlock
+import software.amazon.smithy.rust.codegen.rustlang.rustTemplate
+import software.amazon.smithy.rust.codegen.rustlang.writable
+import software.amazon.smithy.rust.codegen.server.python.smithy.PythonServerCargoDependency
+import software.amazon.smithy.rust.codegen.smithy.CodegenContext
+import software.amazon.smithy.rust.codegen.smithy.RuntimeType
+import software.amazon.smithy.rust.codegen.smithy.RustSymbolProvider
+import software.amazon.smithy.rust.codegen.smithy.expectRustMetadata
+import software.amazon.smithy.rust.codegen.smithy.generators.StructureGenerator
+import software.amazon.smithy.rust.codegen.smithy.rustType
+import software.amazon.smithy.rust.codegen.util.hasTrait
+
+/**
+ * To share structures defined in Rust with Python, `pyo3` provides the `PyClass` trait.
+ * This class generates input / output / error structures definitions and implements the
+ * `PyClass` trait.
+ */
+open class PythonServerStructureGenerator(
+    model: Model,
+    private val codegenContext: CodegenContext,
+    private val symbolProvider: RustSymbolProvider,
+    private val writer: RustWriter,
+    private val shape: StructureShape
+) : StructureGenerator(model, symbolProvider, writer, shape) {
+    private val codegenScope =
+        arrayOf(
+            "pyo3" to PythonServerCargoDependency.PyO3.asType(),
+        )
+
+    override fun renderStructure() {
+        val symbol = symbolProvider.toSymbol(shape)
+        val containerMeta = symbol.expectRustMetadata()
+        if (shape.hasTrait<ErrorTrait>()) {
+            writer.renderPyClassException()
+        } else {
+            writer.renderPyClass()
+        }
+        writer.documentShape(shape, model)
+        val withoutDebug = containerMeta.derives.copy(
+            derives = containerMeta.derives.derives - RuntimeType.Debug + RuntimeType.Clone
+        )
+        containerMeta.copy(derives = withoutDebug).render(writer)
+
+        writer.rustBlock("struct $name ${lifetimeDeclaration()}") {
+            forEachMember(members) { member, memberName, memberSymbol ->
+                renderMemberDoc(member, memberSymbol)
+                renderPyGetterSetter()
+                memberSymbol.expectRustMetadata().render(this)
+                write("$memberName: #T,", symbolProvider.toSymbol(member))
+            }
+        }
+        renderStructureImpl()
+        renderDebugImpl()
+        renderPyO3Methods()
+    }
+
+    private fun renderPyO3Methods() {
+        if (shape.hasTrait<ErrorTrait>() || !accessorMembers.isEmpty()) {
+            writer.renderPyMethods()
+            writer.rustTemplate(
+                """
+                impl $name {
+                    ##[new]
+                    pub fn new(#{bodysignature:W}) -> Self {
```

Comment:
Was there an issue ever cut about this having to be all lowercase?

Reply:
https://github.com/awslabs/smithy-rs/issues/1459